packages/alfa-rectangle/src/rectangle.ts

```diff
@@ -0,0 +1,157 @@
+import { Equatable } from "@siteimprove/alfa-equatable";
+import { Hash, Hashable } from "@siteimprove/alfa-hash";
+import { Serializable } from "@siteimprove/alfa-json";
+
+import * as json from "@siteimprove/alfa-json";
+
+const { max, min } = Math;
+
+/**
+ * @public
+ */
+export class Rectangle
+  implements Equatable, Hashable, Serializable<Rectangle.JSON>
+{
+  public static of(
+    x: number,
+    y: number,
+    width: number,
+    height: number
+  ): Rectangle {
+    return new Rectangle(x, y, width, height);
+  }
+
+  private readonly _x: number;
+  private readonly _y: number;
+  private readonly _width: number;
+  private readonly _height: number;
+
+  private constructor(x: number, y: number, width: number, height: number) {
+    this._x = x;
+    this._y = y;
+    this._width = width;
+    this._height = height;
+  }
+
+  get width() {
+    return this._width;
+  }
+
+  get height() {
+    return this._height;
+  }
+
+  get top() {
+    return this._y;
+  }
+
+  get left() {
+    return this._x;
+  }
+
+  get right() {
+    return this._x + this._width;
+  }
+
+  get bottom() {
+    return this._y + this._height;
+  }
```

Comment:
**Issue:** these should handle negative height/width. And we should probably provide x/y getters also.
See the [DOMRect API](https://developer.mozilla.org/en-US/docs/Web/API/DOMRect).